store/sqlstore/session_store.go

```diff
@@ -52,19 +52,15 @@ func (me SqlSessionStore) Save(session *model.Session) (*model.Session, *model.A
 
 	session.PreSave()
 
-	tcs := me.Team().GetTeamsForUser(session.UserId)
```

Comment:
This was made this way to get the Teams for the user while the session was inserted. I would expect to maintain that concurrency here. Here you should use the explicit async approach, with the channel and the gorouting (see the example in the PR)